### PR TITLE
Add manage subscription implementation

### PR DESCRIPTION
**Pull request by anuruddhal**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/APISubscriptionDAOImpl.java [hunk 1]
```diff
@@ -227,16 +230,32 @@ public Subscription getAPISubscription(String subscriptionId) throws APIMgtDAOEx
...
+                "AND API.UUID = SUBS.API_ID AND API.PROVIDER = ? " +
+                "AND SUBS.SUB_STATUS NOT IN (?,?)";
+        try (Connection conn = DAOUtil.getConnection();
+             PreparedStatement ps = conn.prepareStatement(getSubscriptionsByAPISql)) {
+            ps.setString(1, username);
+            try (ResultSet rs = ps.executeQuery()) {
+                return createSubscriptionsFromResultSet(rs);
+            }
+        } catch (SQLException e) {
+            throw new APIMgtDAOException(e);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/APISubscriptionDAOImpl.java]: log and thwor the exception

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/SubscriptionsApiService.java [hunk 2]
```diff
@@ -1,24 +1,8 @@
 package org.wso2.carbon.apimgt.rest.api.publisher;
 
-import org.wso2.carbon.apimgt.rest.api.publisher.*;
-import org.wso2.carbon.apimgt.rest.api.publisher.dto.*;
-
-import org.wso2.msf4j.formparam.FormDataParam;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/SubscriptionsApiService.java]: was these imports not used before?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/SubscriptionsApiServiceImpl.java [hunk 3]
```diff
@@ -70,12 +68,18 @@ public Response subscriptionsBlockSubscriptionPost(String subscriptionId, String
...
-            List<Subscription> subscriptionList = apiPublisher.getSubscriptionsByAPI(apiId);
-            SubscriptionListDTO subscriptionListDTO = MappingUtil.fromSubscriptionListToDTO(subscriptionList, limit,
-                    offset);
-            return Response.ok().entity(subscriptionListDTO).build();
+            if (StringUtils.isNotEmpty(apiId)) {
+                subscriptionList = apiPublisher.getSubscriptionsByAPI(apiId);
+                SubscriptionListDTO subscriptionListDTO = MappingUtil.fromSubscriptionListToDTO(subscriptionList, limit,
+                        offset);
+                return Response.ok().entity(subscriptionListDTO).build();
+            }else{
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/SubscriptionsApiServiceImpl.java]: slight formattin gissue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/SubscriptionsApiService.java (hunk 2 above)
Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/gen/java/org/wso2/carbon/apimgt/rest/api/publisher/SubscriptionsApiService.java]: These are autogenerate code. SubscriptionDTO object was changed and these are no longer used.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 4]
```diff
@@ -138,15 +138,22 @@ public APIPublisherImpl(String username, ApiDAO apiDAO, ApplicationDAO applicati
...
      *
-     * @param providerId if of the provider
+     * @param providerName username of the provider
      * @return {@code Set<String>}
      * @throws APIManagementException if failed to get subscribed APIs of given provider
      */
     @Override
-    public Set<String> getSubscribersOfProvider(String providerId) throws APIManagementException {
-        return null;
+    public List<Subscription> getSubscribersOfProvider(int offset, int limit, String providerName)
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: Offset and limit are not added in the comment section. Shall we add those?

**Pull request merged**